jail/20230308.md

```diff
@@ -0,0 +1,176 @@
+# FreeBSD Jail Call | 20230308
+
+These calls take place each Wednesday at 10AM Pacific barring conflicts with community events and holidays. The focus alternates between Developers and Production Users each week The [Zoom meeting linkis](https://us06web.zoom.us/j/85638209502?pwd=WEtHK0llaWZmTWtmTnhPUlVqRmlGUT09) available and you are welcome to contact Antranig V. (antranigv@freebsd dot am) and Michael Dexter (editor@callfortesting dot org) with questions, concerns, and to be added or removed from the invitation mailing list. This regular call began on 2023-03-08.
+
+**2023-03-08 Production Users**
+Attendees: Phillip V, Michael D, Alan S, Antranig V, Dave C, Igor A, Phillip V, Goran M, Dan L, Dominik J.
+
+Welcome!
+
+## Introductions
+
+**Alan Somers**: Developer, been using `iocage` and `ezjail` before that. Disappointed that `iocage` is under-maintained. Concerned that Bastille and pot are making: Written in sh, making it hard to debug, resulting in zero test suite.
```

Comment:
@asomers Just a remark: Even though pot is currently written in sh (we are planning to change that in the future), it has a basic test suite consisting of:

1. shellcheck for syntax checking
2. unit tests using shunit
3. A CI test suite

All these are integrated with the github repo (e.g., https://github.com/bsdpot/pot/blob/master/.github/workflows/ci.yml).

Did you try to reach out to the project (or maybe @pizzamig personally) to discuss?

Reply:
Sorry; the notes came out a little too harsh.  I mean that both Bastille and pot are written in sh, making it hard to debug and scale, and that *bastille* had zero test suite.  I saw that Pot had one, though I haven't dug into the details.  What language are you planning to rewrite into?

Reply:
Just to be clear, this is not a *factually correct* document, but rather notes of our call.

We are, however, open to modify the doc, i.e. errata notes from the community 😄

Reply:
@antranigv I understand - this I wasn't on the call/not aware it happened, this seemed like the best place to comment ;) (as it would reach all those who participated)

@asomers We were considering rust, since I had some positive experience with it over the last few years (including writing pot support for sccache) and @pizzamig likes it too. Also, sysutils/potnet is already written in rust anyway, so we won't add more dependencies. It's not decided yet though.

Reply:
Oh, that would be awesome!  I'm a big fan of Rust.  Let me know if you want any help.

Reply:
@asomers I'll keep you posted, as soon as there is basic progress

Reply:
Hi everyone. I'm late, but I hope not too late.
Ideally, we would like to have some sort of `jaild` daemon, to have a better control of the various states of a jail.
I'm not 100% sold to Rust, because we would need to have libjail as a crate. The problem is that triplets in rust don't contain FreeBSD release information (I've broken the libc create already once 😂 )
So C is also on the table, as `jaild` could even be part of FreeBSD eventually.
I've just requested to join next meetings so, we can speak about it